Mini challenge 3

DaniconD opened the pull request:
## Description
Deliverable terminado hasta el punto 3

## Type of Change
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactor (non-breaking refactor which doesn't change functionality)


## How should this be tested?
Al escribir en el buscador se despliegan videos relacionados con la busqueda, los cuales pueden ser clickeados y muestran informacion del video junto con un reproductor

## Any background context you want to provide?
No pude terminar todos los puntos pero dado que lo necesitaba complete puntos de otros deliverables como agregar routers y global context a mi proyecto

## What is the deliverable?
Deliverable 3
Mini-Challenge 3: Hooks

DaniconD committed:
feat: youtube api conexion

Ahora puede usarse un fetch para obtener items de la api de youtube y rendereizar videos con elllos

DaniconD committed:
feat: search field functionality

Ahora es porsible buscar resultados de la API con el input del header

DaniconD committed:
style: video detail view

Añadi el layout y estilos del componente video detail view

DaniconD committed:
feat: video details

al dar click a uno de los videos se abre una pagina con el video, sus datos y una lista de videos

Topi99 commented on src/api/Youtube.js:
```diff
@@ -0,0 +1,73 @@
+import Single from '../components/List/Single';
+import { useSearch } from '../context/SearchContext';
+
+// import axios, { Axios } from "axios";
+// const KEY = 'AIzaSyCEseedJLKU_b4L_xqUVrfmeCYOR3LbCWY';
+
+// export default axios.create({
+//     baseURL: 'https://www.googleaplis.com/youtube/v3/',
+//     params: {
+//         part: 'snippet',
+//         maxResults: 5,
+//         key: KEY
+//     }
+// })
```
Siempre intenta borrar todo el código comentado, ya que no es considerado buena práctica tener código sin usar en producción.

Topi99 commented on src/api/useFetchSingle.js:
```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from 'react';
+function useFetchSingle(id) {
```
¿Este código es usado en algún componente? Solo veo un uso, pero el código está comentado. ¿Qué fue lo que hizo falta para que funcionara este hook?

Topi99 commented on src/components/Video/VideoDetailView.js:
```diff
@@ -0,0 +1,64 @@
+          src={`https://www.youtube.com/embed/${id}`}
+          title="video"
+        ></iframe>
+        <h3>Titulo</h3>
+        <p>
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec
+          placerat porttitor arcu mollis varius. Duis at orci augue. Suspendisse
+          quis varius ligula.
+        </p>
```
¿Tuviste algún impedimento o bloqueo al implementar los detalles del video y los videos recomendados?

Topi99 reviewed (changes requested):
Hola @DaniconD, dejo un par de comentarios y dudas. Si algo no está claro o tienes preguntas/impedimentos/bloqueos con tus proyectos y/o las sesiones del apprenticeship, no dudes en contactarme.

DaniconD committed:
fix: video details uptae

Ahora se muestran los detalles de los videos junto con una lista de relacionados
